control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -0,0 +1,335 @@
+		return true, true, nil
+	}
+	if peeringAcceptor.Status.Secret.Key != peeringAcceptor.Spec.Peer.Secret.Key {
+		return true, false, nil
+	}
+
+	// TODO(peering): remove this when validation webhook exists.
+	if peeringAcceptor.Status.Secret.Backend != peeringAcceptor.Spec.Peer.Secret.Backend {
+		return false, false, errors.New("PeeringAcceptor backend cannot be changed")
+	}
+	// Compare the existing secret hash.
+	// Get the secret specified by the status, make sure it matches the status' secret.latestHash.
+	if existingSecret != nil {
+		existingSecretHashBytes := sha256.Sum256(existingSecret.Data[peeringAcceptor.Status.Secret.Key])
+		existingSecretHash := hex.EncodeToString(existingSecretHashBytes[:])
+		if existingSecretHash != peeringAcceptor.Status.Secret.LatestHash {
+			return true, false, nil
+		}
+
+	}
+	return false, false, nil
+}
+
+// updateStatusError updates the peeringAcceptor's secret in the status.
+func (r *PeeringAcceptorController) updateStatus(ctx context.Context, peeringAcceptor *consulv1alpha1.PeeringAcceptor, resp *api.PeeringGenerateTokenResponse) error {
+	peeringAcceptor.Status.Secret = &consulv1alpha1.SecretStatus{
+		Name:    peeringAcceptor.Spec.Peer.Secret.Name,
+		Key:     peeringAcceptor.Spec.Peer.Secret.Key,
+		Backend: peeringAcceptor.Spec.Peer.Secret.Backend,
+	}
+
+	peeringTokenHash := sha256.Sum256([]byte(resp.PeeringToken))
+	peeringAcceptor.Status.Secret.LatestHash = hex.EncodeToString(peeringTokenHash[:])
+
+	peeringAcceptor.Status.LastReconcileTime = &metav1.Time{Time: time.Now()}
+	peeringAcceptor.Status.ReconcileError = &consulv1alpha1.ReconcileErrorStatus{
+		Error:   pointerToBool2(false),
+		Message: pointerToString(""),
+	}
+	err := r.Status().Update(ctx, peeringAcceptor)
+	if err != nil {
+		r.Log.Error(err, "failed to update PeeringAcceptor status", "name", peeringAcceptor.Name, "namespace", peeringAcceptor.Namespace)
+	}
+	return err
+}
+
+// updateStatusError updates the peeringAcceptor's ReconcileError in the status.
+func (r *PeeringAcceptorController) updateStatusError(ctx context.Context, peeringAcceptor *consulv1alpha1.PeeringAcceptor, reconcileErr error) error {
+	peeringAcceptor.Status.ReconcileError = &consulv1alpha1.ReconcileErrorStatus{
+		Error:   pointerToBool2(true),
+		Message: pointerToString(reconcileErr.Error()),
+	}
+
+	peeringAcceptor.Status.LastReconcileTime = &metav1.Time{Time: time.Now()}
+	err := r.Status().Update(ctx, peeringAcceptor)
+	if err != nil {
+		r.Log.Error(err, "failed to update PeeringAcceptor status", "name", peeringAcceptor.Name, "namespace", peeringAcceptor.Namespace)
+	}
+	return err
+}
+
+// getExistingSecret gets the K8s secret specified, and returns a bool indicating whether it exists, and the secret
+// itself. It returns nil if it doesn't exist.
+func (r *PeeringAcceptorController) getExistingSecret(ctx context.Context, name string, namespace string) (bool, *corev1.Secret, error) {
```

Comment:
Could we check if secret is nil to check if it exists instead of returning an additional boolean?

Reply:
Yup! Changed!